src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/HotReload/IProjectHotReloadSession.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.HotReload
+{
+    public interface IProjectHotReloadSession
+    {
+        bool IsSupported { get; }
+
+        Task StartSessionAsync(CancellationToken cancellationToken);
+
+        Task StopSessionAsync(CancellationToken cancellationToken);
+
+        Task ApplyChangesAsync(CancellationToken cancellationToken);
+
+        Task<bool> ApplyLaunchVariablesAsync(IDictionary<string, string> envVars, string configuration, CancellationToken cancellationToken);
```

Comment:
Does this need a mutable dictionary?

Reply:
We may need to merge the value of existing environment variables with the Hot Reload values. Either we mutate the dictionary, or we take in the current set of environment variables as an immutable object, produce a new immutable object with the changes, and then let the caller merge that back in.